### PR TITLE
[NFC] Adjust several lit tests to use opaque pointers.

### PR DESCRIPTION
**Pull request by jcranmer-intel**
These tests generally require adjustments to their check lines to match opaque pointer output, or extra bitcasts, or the like correctly. For the function pointer tests in particular, some adjustments to the input IR have been made to minimize the impact on existing checks.

### CODE REVIEW

File: test/extensions/INTEL/SPV_INTEL_function_pointers/bitcast.ll [hunk 1]
```diff
@@ -37,8 +39,8 @@ define dso_local spir_func signext i8 @foo(i8 signext %0) #0 {
 
 ; Function Attrs: noinline nounwind optnone
 define dso_local spir_func void @bar() #0 {
-  %1 = alloca i32 (i32)*, align 4
-  store i32 (i32)* bitcast (i8 (i8)* @foo to i32 (i32)*), i32 (i32)** %1, align 4
+  %1 = bitcast i8 (i8)* @foo to i32 (i32)*
+  %2 = call i32 %1(i32 0)
```
Comment (MrSidims) [inline test/extensions/INTEL/SPV_INTEL_function_pointers/bitcast.ll:43]: @AlexeySachkov may I ask you to take a look if you are OK with the test change

File: test/layout.ll [hunk 2]
```diff
@@ -124,7 +123,7 @@ target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:2
 target triple = "spir"
 
 @v = addrspace(1) global [2 x i32] [i32 1, i32 2], align 4
-@s = addrspace(1) global i32 addrspace(1)* getelementptr inbounds ([2 x i32], [2 x i32] addrspace(1)* @v, i32 0, i32 0), align 4
```
[the remark this answers is not in the document]
Comment (jcranmer-intel) [reply test/layout.ll:127]: The reason for this change is that `getelementptr @v, i32 0, i32 0` in opaque pointer mode becomes just `@v`, which means the scavenged type of `@s` changes from `i32**` to `[2 x i32]**`. Replacing the final `i32 0` with an `i32 1` keeps the scavenged type as `i32**`

File: test/extensions/INTEL/SPV_INTEL_function_pointers/bitcast.ll (hunk 1 above)
Comment (AlexeySachkov) [reply test/extensions/INTEL/SPV_INTEL_function_pointers/bitcast.ll:43]: It looks like this IR doesn't match OpenCL C snippet in the comments above. I'm fine with the IR change, but we need to align OpenCL C snippet then to add call there.

**Comment by MrSidims** [comment]
test failures are unrelated

**Review by MrSidims — APPROVED** [review]

**Pull request merged**